### ISSUE TITLE
Files created in virtual file system

### ISSUE THREAD

**Issue by pyRobShrk** [pyodide/pyodide#1130]
When files are created in the working directory (virtual) with **open**() or **pandas.DataFrame.to_csv()** or **imageio**, there is no error thrown, but the files are not readily available in the browser without some conversion using StringIO. Ideally any new files created in the virtual file system by Python would be automatically downloaded in the web browser. As these writing methods also allow a file-like object, another idea might be to have a **pyodide.file()** object in the Python API that could behave as I'm describing - when closed/saved the file is downloaded in the web browser.

**Comment by hoodmane** [comment]
> any new files created in the virtual file system by Python would be automatically downloaded in the web browser

I don't think this is a good idea, most files written you probably don't want downloaded. I agree it would be a good idea to offer some better support for moving files into and out of the virtual filesystem.

> when closed/saved the file is downloaded in the web browser

You mean into the native downloads folder? Maybe there could be a context manager for this behavior. I don't know that much about Python file system APIs, but presumably there is a "file handle closed" hook somewhere. I think though that an explicit API would be preferable to an implicit one.

**Comment by ryanking13** [comment]
I am not sure whether the "downloaded" here means downloading the file into the native filesystem, but we now have [Native File system API](https://pyodide.org/en/stable/usage/file-system.html#experimental-using-the-native-file-
system-in-the-browser) for mounting the native FS to Pyodide FS. So I'll close this as resolved.
